data_steward/tools/clean_project_datasets.py

```diff
@@ -58,31 +51,59 @@ def run_deletion(project_id, name_substrings):
     :param name_substrings: Identifies substrings that help identify datasets to delete
     :return:
     """
+    # make the developer running this script approve the environment.
+    msg = (f'This will remove datasets from the `{project_id}` '
+           f'environment.\nAre you sure you with to proceed?  '
+           f'[Y/y/N/n]:  ')
+
+    LOGGER.info(msg)
+    proceed = input(msg)
+
+    LOGGER.info(f'User entered: "{proceed}"')
+
+    if proceed.lower() != 'y':
+        LOGGER.info(f'User requested to exit the deletion script.\n'
+                    f'Exiting clean_project_datasets script now.')
+        return
+
+    LOGGER.info('Continuing with dataset deletions...')
+
+    client = bq.get_client(project_id)
+
     all_datasets = [
-        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+        dataset.dataset_id for dataset in list(client.list_datasets())
     ]
-    datasets_with_substrings = get_datasets_with_substrings(
-        all_datasets, name_substrings)
-    logging.info(f'Datasets marked for deletion: {datasets_with_substrings}')
-    logging.info('Proceed?')
-    response = get_response()
-    if response == "Y":
-        delete_datasets(project_id, datasets_with_substrings)
-    elif response.lower() == "n":
-        logging.info("Aborting deletion")
-
-
-# Make sure user types Y to proceed
-def get_response():
-    """Return input from user denoting yes/no"""
-    prompt_text = 'Please press Y/n\n'
-    response = input(prompt_text)
-    while response not in ('Y', 'n', 'N'):
-        response = input(prompt_text)
-    return response
 
+    datasets_with_substrings = [
+        dataset for dataset in all_datasets for substring in name_substrings
+        if substring in dataset
+    ]
 
-if __name__ == '__main__':
+    LOGGER.info(f'{len(datasets_with_substrings)} Datasets marked for '
+                f'deletion in project `{project_id}`: ')
+    for dataset in datasets_with_substrings:
+        LOGGER.info(f'\t{dataset}')
+
+    msg = (f'After reviewing datasets, proceed?\nThis action '
+           f'cannot be reversed.\n'
+           f'[Y/y/N/n]:  ')
+    LOGGER.info(msg)
+    response = input(msg)
```

Comment:
This doesn't show the datasets to review, so I expanded the `msg`. However, this is not a required change since there is a warning for setting console log in the beginning.
```suggestion
    msg = (f'{len(datasets_with_substrings)} Datasets marked for '
           f'deletion in project `{project_id}`: \n' +
           '\n'.join(f'\t{dataset}' for dataset in datasets_with_substrings) +
           f'\nAfter reviewing datasets, proceed?\nThis action '
           f'cannot be reversed.\n'
           f'[Y/y/N/n]:  ')
    response = input(msg)
```

Reply:
I'm not making this change.  We're already repeating the messages via logs and input messages.  It feels like we are trying to force this to be an always command line tool that needs to have someone watching it.